File: Wms.Api/Dto/StockIn/StockInCreateUpdate/StockInCreateUpdateDto.cs
namespace Wms.Api.Dto.StockIn.StockInCreateUpdate
{
    public class StockInCreateUpdateDto
    {
        public required string Number { get; set; }
        public required string SellerInfo { get; set; }
        public required string Purchaser { get; set; }
        public required string Location { get; set; }
        public DateTime DateOfPurchase { get; set; }
        public Guid WarehouseId { get; set; }
        public ICollection<StockInItemCreateUpdateDto>? StockInItems { get; set; }
    }
}

File: Frontend/src/interfaces/v12/stockIn/stockInCreateUpdate/stockInCreateUpdateDto.ts
import { guid } from "types/guid";
import { StockInItemCreateUpdateDto } from "./stockInItemCreateUpdateDto";

export interface StockInCreateUpdateDto {
  sellerInfo: string;
  purchaser: string;
  dateOfPurchase: string;
  warehouseId: guid;
  stockInItems?: StockInItemCreateUpdateDto[];
}
